### ISSUE TITLE
Improve Verbosity of Elasticsearch cluster_block_exception Message Payload

### ISSUE THREAD

**Issue by berglh** [elastic/elasticsearch#28592]
## Environment
Elasticsearch 5.5.1 & 6.1.1

## Proposal
Add the target index name in the response message of the cluster_block_exception

## Justification
Recently I have implemented a new policy of setting older indices as read-only to prevent things such as misdirected `delete_by_query` operations :unamused:. I also run Elastic Curator to close the oldest indices for period of seven days before they're permanently removed from the cluster; in the same job we also mark any index not in the current month or year as read only. We aim for indices with a time range of either one month or one year depending on doc size, volume, field cardinality.

At the start of this month, an older index was closed and more recent indices with the same index pattern prefix as read-only. However; due to an ill thought out process, an upstream data provider sent me data to be indexed in the index that was closed, in this case an index created in 2016.

These old timestamped events entered all of our redis queues and were picked up by our logstash processors. The logstash pipelines were then clogged with documents that were unable to be indexed because they were trying to be written to the closed index.

At first, the visibility of what was causing the blockage was limited in both elasticsearch and logstash logging; so I enabled debug log output on one of the logstash processors. Sure enough, logstash indicated that an event was trying to write to an index that was actually closed and returned the name of the index so I was able to identify the source of the events.

However, after re-opening the index, additional events from the same source were then trying to be written to the now read-only indices. In this case, debug/trace level logs of logstash didn't contain any index information for the 403 error.

So I tested the same operations with the typical usage for the Index API specifying `POST (index_name)/(type)?/(id)?` along with the document content to write/update. The error responses from elasticseach returned the **index** name on the closed index but not on the read only index.

While this wouldn't matter if you knew the target index during the write operation, trying to diagnose pipeline obstructions is difficult in logstash if elasticsearch isn't returning the index name during a blocked write to a read only index.

## Write to Read Only Index
```json
{
  "error": {
    "root_cause": [
      {
        "type": "cluster_block_exception",
        "reason": "blocked by: [FORBIDDEN/5/index read-only (api)];"
      }
    ],
    "type": "cluster_block_exception",
    "reason": "blocked by: [FORBIDDEN/5/index read-only (api)];"
  },
  "status": 403
}
```
## Write to Closed Index
```json
{
  "error": {
    "root_cause": [
      {
        "type": "index_closed_exception",
        "reason": "closed",
        "index_uuid": "o9M_wB0wRGW88l6BgiNJFg",
        "index": "old-index-2017.09"
      }
    ],
    "type": "index_closed_exception",
    "reason": "closed",
    "index_uuid": "o9M_wB0wRGW88l6BgiNJFg",
    "index": "old-index-2017.09"
  },
  "status": 403
}
```

**Comment by adityasrini** [comment]
Would love to work on this if no one is!

**Comment by pallix** [comment]
I have exactly the same issue with the `FORBIDDEN/4/index` error and it would be tremendously helpful.

I have opened a separate issue for logstash https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/888

**Comment by pallix** [comment]
Even doing that does not output anything:

```
PUT /_cluster/settings
{
  "transient": {
    "logger.org.elasticsearch.action.bulk": "DEBUG"
  }
}
```

**Comment by berglh** [comment]
> ```
> PUT /_cluster/settings
> {
>   "transient": {
>     "logger.org.elasticsearch.action.bulk": "TRACE"
>   }
> }
> ```
> 
> shows useful information.

This is fantastically useful, thanks for this.

**Comment by dakrone** [comment]
This has been open for quite a while, and we haven't made much progress on this due to focus in other areas. For now I'm going to close this as something we aren't planning on implementing. We can re-open it later if needed.
